openwisp_utils/base.py

```diff
@@ -3,23 +3,38 @@
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 from model_utils.fields import AutoCreatedField, AutoLastModifiedField
+from openwisp_utils.utils import get_random_key
+from openwisp_utils.validators import key_validator
 
 
-class TimeStampedEditableModel(models.Model):
+class UUIDModel(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class TimeStampedEditableModel(UUIDModel):
     """
     An abstract base class model that provides self-updating
     ``created`` and ``modified`` fields.
     """
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     created = AutoCreatedField(_('created'), editable=True)
     modified = AutoLastModifiedField(_('modified'), editable=True)
 
     class Meta:
         abstract = True
 
 
-class UUIDModel(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-
-    class Meta:
-        abstract = True
+class KeyField(models.CharField):
+    def __init__(self, unique: bool, db_index: bool, help_text: str, *args, **kwargs):
+        fields = {
+            'max_length': 64,
+            'unique': unique,
+            'db_index': db_index,
+            'default': get_random_key,
+            'validators': [key_validator],
+            'help_text': help_text
+        }
+        fields.update(kwargs)
+        super().__init__(*args, **fields)
```

Comment:
Please do the following:

move this field to a new `fields.py` file, since in `base.py` we are keeping the base models and is a good practice to separate logic in different files, plus `fields.py` is common across django packages.
After you do this you'll need to update your patch on django-netjsongraph.

Then update the code of the field as follows:

```python
class KeyField(models.CharField):
    def __init__(self,
                 max_length: int = 64,
                 unique: bool = False,
                 db_index: bool = False,
                 help_text: str = None,
                 *args, **kwargs):
        kwargs.update({
            'default': get_random_key,
            'validators': [key_validator],
        })
        super().__init__(max_length=max_length,
                         unique=unique,
                         db_index=db_index,
                         help_text=help_text,
                         *args, **kwargs)
```

Your version doesn't supply default values and hence initializing the field without an explicit `unique=False` will cause it to raise the exception.